torch/nn/modules/transformer.py

```diff
@@ -311,6 +320,7 @@ def __init__(self, decoder_layer, num_layers, norm=None):
         super(TransformerDecoder, self).__init__()
         torch._C._log_api_usage_once(f"torch.nn.modules.{self.__class__.__name__}")
         self.layers = _get_clones(decoder_layer, num_layers)
+        self.decoder_layer = decoder_layer
```

Comment:
that's the same as self.layers[0] ?

Reply:
Hi, using self.layers[0] may return index error.